src/Collection.ts

```diff
@@ -68,7 +70,7 @@ class Collection extends Resource {
      * @returns array          the new array with replaced items, or a LoadingCollection if any of the array
      *                         elements is still loading.
      */
-  private _mapArrayOfEntityReferences (array: Array<Link>): Array<ResourceInterface> {
+  private _mapArrayOfEntityReferences (array: Array<ItemType>): Array<ItemType> {
```

Comment:
is ItemType the same as Link?
the name suggests that there should be a mapping, probably of one type to another

Reply:
I didn't know how to type that some items could be of store type link